internal/dataplane/sendconfig/error_handling.go

```diff
@@ -0,0 +1,148 @@
+	UID        string
+	Problems   map[string]string
+}
+
+// rawResourceError is a Kong configuration error associated with a Kubernetes resource with Kubernetes metadata stored
+// in raw Kong entity tags.
+type rawResourceError struct {
+	Name     string
+	ID       string
+	Tags     []string
+	Problems map[string]string
+}
+
+// ConfigError is an error response from Kong's DB-less /config endpoint.
+type ConfigError struct {
+	Code      int               `json:"code,omitempty" yaml:"code,omitempty"`
+	Flattened []FlatEntityError `json:"flattened_errors,omitempty" yaml:"flattened_errors,omitempty"`
+	Message   string            `json:"message,omitempty" yaml:"message,omitempty"`
+	Name      string            `json:"name,omitempty" yaml:"name,omitempty"`
+}
+
+// ConfigErrorFields is the structure under the "fields" key in a /config error response.
+type ConfigErrorFields struct {
+}
+
+// FlatEntityError represents a single Kong entity with one or more invalid fields.
+type FlatEntityError struct {
+	Name   string           `json:"entity_name,omitempty" yaml:"entity_name,omitempty"`
+	ID     string           `json:"entity_id,omitempty" yaml:"entity_id,omitempty"`
+	Tags   []string         `json:"entity_tags,omitempty" yaml:"entity_tags,omitempty"`
+	Errors []FlatFieldError `json:"errors,omitempty" yaml:"errors,omitempty"`
+}
+
+// FlatFieldError represents an error for a single field within a Kong entity.
+type FlatFieldError struct {
+	Field string `json:"field,omitempty" yaml:"field,omitempty"`
+	// Message is the error associated with Field for single-value fields.
+	Message string `json:"message,omitempty" yaml:"message,omitempty"`
+	// Messages are the errors associated with Field for multi-value fields. The array index in Messages matches the
+	// array index in the input.
+	Messages []string `json:"messages,omitempty" yaml:"messages,omitempty"`
+}
+
+// parseFlatEntityErrors takes a Kong /config error response body and parses its "fields.flattened_errors" value
+// into errors associated with Kubernetes resources.
+func parseFlatEntityErrors(body []byte, log logrus.FieldLogger) ([]ResourceError, error) {
+	var resourceErrors []ResourceError
+	var configError ConfigError
+	err := json.Unmarshal(body, &configError)
+	if err != nil {
+		return resourceErrors, fmt.Errorf("could not unmarshal config error: %w", err)
+	}
+	for _, ee := range configError.Flattened {
+		raw := rawResourceError{
+			Name:     ee.Name,
+			ID:       ee.ID,
+			Tags:     ee.Tags,
+			Problems: map[string]string{},
+		}
+		for _, p := range ee.Errors {
+			if len(p.Message) > 0 && len(p.Messages) > 0 {
+				log.WithFields(logrus.Fields{
+					"name":  ee.Name,
+					"field": p.Field}).Error("entity has both single and array errors for field")
+				continue
+			}
+			if len(p.Message) > 0 {
+				raw.Problems[p.Field] = p.Message
+			}
+			if len(p.Messages) > 0 {
+				for i, message := range p.Messages {
+					if len(message) > 0 { // TODO how are the nulls treated?
```

Comment:
What do you mean by "nulls"? `null` (null value) in json will be unmarshalled to empty string.

Reply:
Don't recall. I think this is a leftover from my original PoC--maybe something in my test gateway code would produce `null` messages that I thought I needed to handle. Don't know of obvious issues with the messages with the non-PoC implementation (https://github.com/Kong/kong/pull/10161). Subjectively, the logs I saw while building the integration test looked reasonable with the actual implementation, so just dropping this a PoC leftover.